src/components/profile/AccountPlansAndBilling.tsx

```diff
@@ -121,70 +323,253 @@ const AccountPlansAndBilling: React.FC<{ currentAccount: Account }> = ({
-              'Unlimited contact connection',
-              'Email support',
-            ]}
-          />
-        </Flex>
-
-        <SubscriptionDialog
-          isDialogOpen={isDialogOpen}
-          onDialogClose={() => setIsDialogOpen(false)}
-          cancelDialogRef={cancelDialogRef}
-          onSuccessPurchase={subsPurchased}
-          currentSubscription={currentAccount?.subscriptions?.[0]}
-          defaultCoupon={Array.isArray(coupon) ? coupon[0] : coupon}
-        />
+        {billingLoading ? (
+          <Box width="100%">
+            <CustomLoading text="Loading subscription..." />
+          </Box>
+        ) : (
+          <Flex
+            width="100%"
+            flexDirection={{ base: 'column', md: 'row' }}
+            gridGap={2}
+          >
+            <SubscriptionCard
+              subscription={
+                // Use blockchain subscription if no billing subscription exists
+                !billingSubscription
+                  ? currentAccount?.subscriptions?.find(
+                      sub => new Date(sub.expiry_time) > new Date()
+                    )
+                  : undefined
+              }
+              planInfo={getPlanInfo(Plan.PRO)}
+              active={isProCardActive}
+              benefits={[
+                'Everything in Free plus (+)',
+                'Unlimited scheduling groups',
+                'Payments & Invoicing',
+                'Unlimited integrations (Google calendar, iCloud, Office 365 and WebDAV)',
+                'Unlimited QuickPolls',
+                'Unlimited meeting types - Free & Paid',
+                '24/7 priority support',
+              ]}
+              badge={activeBadge}
+              expiryText={currentExpiryText}
+              primaryCtaLabel={
+                hasActiveSubscription
+                  ? billingSubscriptionResponse?.payment_provider ===
+                    PaymentProvider.STRIPE
+                    ? undefined
+                    : 'Extend Plan'
+                  : 'Subscribe to Pro'
+              }
+              onClick={handlePrimaryButtonClick}
+              secondaryCtaLabel={
+                hasActiveSubscription &&
+                billingSubscriptionResponse?.payment_provider ===
+                  PaymentProvider.STRIPE
+                  ? 'Manage Subscription'
+                  : undefined
+              }
+              onSecondaryCta={
+                hasActiveSubscription &&
+                billingSubscriptionResponse?.payment_provider ===
+                  PaymentProvider.STRIPE
+                  ? () => void handleManageSubscription()
+                  : undefined
+              }
+              secondaryCtaLoading={manageSubscriptionLoading}
+              tertiaryCtaLabel={
+                hasActiveSubscription &&
+                billingSubscriptionResponse?.payment_provider !==
```

Comment:
move to a single variable and pass as a props to SubscriptionCard to let that dynamically handle rendering theses data